skills_DS/react/src/components/Upload.js

```diff
@@ -20,16 +20,36 @@ const Upload = () => {
     }
   }
 
+  const checkResumeProcessing = () => {
+    try {
+      axios.get("/api/get-profile", {headers: {"X-CSRFTOKEN": Cookies.get("csrftoken")}}).then((response) => {
+        if (!response.data.success.resume_processing) {
+          window.location.href = "profile/";
+        } else {
+          setTimeout(checkResumeProcessing, 2000);
+        }
+      });
+    } catch (err) { }
+  }
+
   const handleSubmission = () => {
     try {
-      const form_data = new FormData()
+      const form_data = new FormData();
 
-      form_data.append("file", selectedFile)
+      form_data.append("file", selectedFile);
 
-      axios.post("/api/fileupload", form_data, {headers: {"X-CSRFTOKEN": Cookies.get("csrftoken"), "Content-Disposition": `attachment; filename=${selectedFile.name}`, "Content-Type": "multipart/form-data"}}).catch()
+      axios.post("/api/fileupload", form_data, {
+        headers: {
+          "X-CSRFTOKEN": Cookies.get("csrftoken"),
+          "Content-Disposition": `attachment; filename=${selectedFile.name}`,
+          "Content-Type": "multipart/form-data"
+        }
+      }).then((response) => {
+        setButtonClicked(true);
+        checkResumeProcessing();
+      }).catch(err => console.error(err));
 
-      setButtonClicked(true)
-    } catch (err) {}
+    } catch (err) { }
```

Comment:
try catch outside seems unnecessary now with .catch

Reply:
Will be resolve in `feat-refactor`